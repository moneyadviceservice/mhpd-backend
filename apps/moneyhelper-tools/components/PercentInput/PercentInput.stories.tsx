import { StoryFn } from '@storybook/react';
import { PercentInput } from './PercentInput';

const StoryProps = {
  title: 'Components/TOOLS/PercentInput',
  component: PercentInput,
};

const Template: StoryFn = (args) => <PercentInput {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'label',
  id: 'id',
  name: 'name',
  defaultValue: '',
  onChange: () => {},
};

export default StoryProps;
