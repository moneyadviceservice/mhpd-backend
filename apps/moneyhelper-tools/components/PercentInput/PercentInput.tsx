import { NumberFormatValues } from 'react-number-format';
import type { Props as NumberInputProps } from '../NumberInput';
import { NumberInput } from '../NumberInput';
import { twMerge } from 'tailwind-merge';

export type PercentInputProps = NumberInputProps & {
  inputClassName?: string;
  inputBackground?: string;
};

export const PercentInput = ({
  inputClassName,
  inputBackground,
  ...rest
}: PercentInputProps) => {
  const MAX_PERCENT_VALUE = 100;

  const minMaxNumberCheck = ({ floatValue }: NumberFormatValues) => {
    if (!floatValue) {
      return true;
    }
    return floatValue <= MAX_PERCENT_VALUE && floatValue >= 0;
  };

  return (
    <div className={twMerge('flex', inputClassName)}>
      <NumberInput
        isAllowed={minMaxNumberCheck}
        className={twMerge(
          'border-gray-400 border-t border-b border-r-0 border-l-1 rounded-l',
          inputBackground,
        )}
        {...rest}
      />
      <span
        aria-hidden
        className="bg-gray-100 px-3 h-[49px] rounded-r border-gray-400 border border-solid justify-center flex items-center"
      >
        %
      </span>
    </div>
  );
};
