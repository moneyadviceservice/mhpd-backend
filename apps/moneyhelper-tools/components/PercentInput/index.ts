export { PercentInput } from './PercentInput';
export type { PercentInputProps } from './PercentInput';
