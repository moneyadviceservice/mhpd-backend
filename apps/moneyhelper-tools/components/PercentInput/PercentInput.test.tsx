import { render } from '@testing-library/react';
import { PercentInput } from './PercentInput';

describe('PercentInput component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PercentInput
        id="test-id"
        name="test-name"
        defaultValue=""
        onChange={() => {}}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
