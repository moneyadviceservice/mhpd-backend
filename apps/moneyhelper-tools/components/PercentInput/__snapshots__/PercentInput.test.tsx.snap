// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentInput component renders correctly 1`] = `
<div
  class="flex"
>
  <input
    class="tool-field w-full border pl-2 ml-0 h-auto focus:outline-purple-700 focus:shadow-focus-outline border-gray-400 border-t border-b border-r-0 border-l-1 rounded-l"
    data-testid="number-input"
    id="test-id"
    inputmode="numeric"
    name="test-name"
    type="text"
    value=""
  />
  <span
    aria-hidden="true"
    class="bg-gray-100 px-3 h-[49px] rounded-r border-gray-400 border border-solid justify-center flex items-center"
  >
    %
  </span>
</div>
`;
